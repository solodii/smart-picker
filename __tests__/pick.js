var createPicker = require('../src/index').createPicker;
var mocks = require('../mocks');

describe('pick', function() {
  var picker = createPicker(mocks.schema);

  it('should return an object if target is not an object', function() {
    expect(picker.pick()).toEqual({});
    expect(picker.pick(null)).toEqual({});
    expect(picker.pick(0)).toEqual({});
    expect(picker.pick('')).toEqual({});
  });

  it('should pick queried object properties', function() {
    expect(
      picker.pick(mocks.target)
    ).toMatchSnapshot();
  });
});
