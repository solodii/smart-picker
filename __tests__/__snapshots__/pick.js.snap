// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pick should pick queried object properties 1`] = `
Object {
  "a_alias": "a_prop",
  "b": Object {
    "c": Object {
      "d_alias": "d_prop",
    },
    "e": Object {
      "f": Object {
        "g": "g_prop",
        "h": "h_prop",
      },
    },
  },
  "func": Object {
    "length": 3,
  },
}
`;
